Updates to variable-based encoding (#1)
ax3l/openPMD-standard: STANDARD.md

    - allowed values:
      - `fileBased` (multiple files)

>       - `groupBased` (one file)

Reviewer: ```suggestion
      - `groupBased` (one file; iteration use groups in that file)
```

>       - `fileBased` (multiple files)
>       - `groupBased` (one file)

Reviewer: ```suggestion
      - `fileBased` (multiple files; one iteration per file)
      - `groupBased` (one file; iteration use groups in that file)
```

>       - `variableBased` (one file with internal encoding for iterations, if supported by the data format)

Reviewer: ```suggestion
      - `variableBased` (one file; if the data format supports to store multiple iterations in the same variables and attributes)
```

        - `filename_%T.h5` (without file system directories)
      - for `groupBased`: (fixed value)
        - `/data/%T/` (must be equal to and encoded in the `basePath`)
      - for `variableBased`: (fixed value)
        - data-format internal convention
        - *slowest varying index* of data

### `variableBased` Encoding of Iterations

> In order to correlate openPMD iterations with an index of data-format internal updates/steps or an index in the slowest varying dimension of an array, the iteration base path (default: path `/default`) must contain an additional variable once `variableBased` is chosen for `iterationEncoding`:

Reviewer: It's `/default`? That is a but curious...

Author: don't know what went wrong here, I'll revert it

Author: Ah, I confused `default` with `data`. At least got the first letter right there..

    - advice to implementers: an openPMD iteration might be spread over multiple updates/steps, but not vice versa.
                              In such a scenario, an individual openPMD record's update/step must appear exactly once per iteration.

Notes:

> * In implementations without support for IO steps, the variable-based encoding of iterations may still be used for storage of a single iteration.

Reviewer: In these points, the new wording using `IO steps` might be confusing?

Author: I've reworded that to `support for storing multiple versions of datasets/attributes`, which I think is more explicit anyway.

  In that case, the `snapshot` attribute is optional and defaults to zero (0).
* In implementations with support for IO steps, the `snapshot` attribute may optionally be used in group-based encoding to associate openPMD iterations with IO steps.
  In group-based encoding, there is still only one instance of this attribute globally (`/data/snapshot`).
  In consequence, the attribute shall only be written if modifiable attributes are supported by the implementation.

Required Attributes for the `basePath`
--------------------------------------

> In addition to holding information about the iteration, each series of files  (`fileBased`), series of groups (`groupBased`) or internally encoded iterations (`variableBased`) should have attributes that describe the current time and the last time step.

Reviewer: iterations (or snapshots) are not necessarily in time :)
```suggestion
In addition to holding information about the iteration, each series of files  (`fileBased`), series of groups (`groupBased`) or internally encoded iterations (`variableBased`) should have attributes that describe the current time and the last step.
```
